mono/metadata/w32socket-unix.c

```diff
@@ -625,6 +627,82 @@ mono_w32socket_transmit_file (SOCKET hSocket, gpointer hFile, guint32 nNumberOfB
 	return TransmitFile (hSocket, hFile, nNumberOfBytesToWrite, nNumberOfBytesPerSend, lpOverlapped, lpTransmitBuffers, dwReserved);
 }
 
+gint
+mono_w32socket_set_blocking (SOCKET socket, gboolean blocking)
```

Comment:
I like this change. It makes explicit that we're not trying to support ioctlsocket.